roles/schema_registry/tasks/health_check.yml

```diff
@@ -35,7 +65,29 @@
   until: result2.status == 200
   retries: 30
   delay: 5
-  when: schema_registry_authentication_type == 'mtls'
+  when:
+    - schema_registry_authentication_type == 'mtls'
+    - not (rbac_enabled or schema_registry_oauth_enabled)
+  ignore_errors: true
+
+- name: Wait for API to return 200 - mTLS
+  uri:
+    url: "{{schema_registry_http_protocol}}://{{ hostvars[inventory_hostname]|confluent.platform.resolve_hostname }}:{{schema_registry_listener_port}}/subjects"
+    status_code: 200
+    validate_certs: false
+    client_cert: "{{ schema_registry_cert_path if ssl_provided_keystore_and_truststore else certs_chain }}"
+    client_key: "{{schema_registry_key_path}}"
+    headers:
+      Content-Type: application/json
+      Authorization: "Bearer {{ authorization_token }}"
+  register: result2_oauth
+  until: result2_oauth.status == 200
+  retries: 30
+  delay: 5
+  when:
+    - schema_registry_authentication_type == 'mtls'
+    - rbac_enabled or schema_registry_oauth_enabled
```

Comment:
when we have both mtls and oauth true then this request is send both bearer token and certs. In this case i am curious does the server validate both or only one set is correct then it authenticates ?

Reply:
it validates both in this case